src/gui_common/dialogs/FilterWindow.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections.Generic;
+using Godot;
+
+public class FilterWindow : CustomConfirmationDialog
+{
+    // Paths for nodes
+    [Export]
+    public NodePath FiltersContainerPath = null!;
+
+    private readonly List<Filter> filters = new();
+    /// <summary>
+    ///   If redraw is needed.
+    /// </summary>
+    private bool dirty;
+
+    private PackedScene filterScene = GD.Load<PackedScene>("res://src/gui_common/dialogs/FilterLine.tscn");
```

Comment:
I don't know if this is okay, but we mostly load the scenes in _ready

Reply:
Well, it is okay in the sense that it is functional. But I guess it falls under the guideline considerations on nodes, which advise against what I did: 

Avoid using a constructor to setup Godot resources, usually Node derived types should mostly do Godot Node related, constructor-like operations entirely in _Ready. [...]

Reply:
Yeah, this is exactly what the style guide is trying to say, scenes should be loaded in _Ready.

Reply:
Alright, thanks for the confirmation; I've changed it. I must say though I find this point not very clear, as I would probably not known what it meant without knowing what I was looking for. An example or two could be useful there I'd think.